File: elixir/gigasecond/lib/gigasecond.ex
defmodule Gigasecond do
  @doc """
  Calculate a date one billion seconds after an input date.
  """
  @spec from({{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}) ::
          :calendar.datetime()

  def from(erlang_date_time = {{_year, _month, _day}, {_hours, _minutes, _seconds}}) do
    gigasecond = :math.pow(10, 9) |> round
    erlang_date_time |> NaiveDateTime.from_erl!() |> NaiveDateTime.add(gigasecond) |> NaiveDateTime.to_erl()
  end
end

File: erlang/gigasecond/src/gigasecond.erl
-module(gigasecond).

%% -import(calendar,[datetime_to_gregorian_seconds/1,gregorian_seconds_to_datetime/1]).

-export([from/1]).


from(_From = {Y,D,M}) ->
    from({{Y,D,M},{0,0,0}});
from(From = {{_Y,_D,_Mo},{_H,_Mi,_S}}) ->
    DateInSeconds = calendar:datetime_to_gregorian_seconds(From),
    NewDateInSeconds = DateInSeconds + 1000000000,
    calendar:gregorian_seconds_to_datetime(NewDateInSeconds).

